src/WebpackAssetsManifest.js

```diff
@@ -544,15 +546,34 @@ class WebpackAssetsManifest
    */
   handleNormalModuleLoader(loaderContext, module)
   {
-    const { emitFile } = loaderContext;
+    const { emitFile, rootContext } = loaderContext;
 
     loaderContext.emitFile = (name, content, sourceMap) => {
       if ( ! this.assetNames.has( name ) ) {
-        const originalName = path.join(
+        let originalName = path.join(
           path.dirname(name),
           path.basename(module.userRequest)
         );
 
+        if (this.options.usePathBasedKeys) {
+          const options = loaderUtils.getOptions(loaderContext);
+
+          // If the option is enabled, attempt to prefix the keys in our
+          // manifest with [path]
+          const pathVariable = loaderUtils.interpolateName(
+            loaderContext,
+            '[path]',
+            // Use any custom context we've been given, otherwise just use
+            // the root context
+            { context: options && options.context ? options.context : rootContext }
+          );
+
+          // Don't prefix any keys that already start with [path]
+          if (!originalName.startsWith(pathVariable)) {
+            originalName = path.join(pathVariable, originalName);
+          }
+        }
```

Comment:
What's the reason for conditionally adding the resolved `pathVariable` to `originalName`? Shouldn't the path always be added? Or can we simplify resolving the path seperately, like this:

```suggestion
        let pathName = path.dirname(name);
        if (this.options.usePathBasedKeys) {
          // Prefix the keys in our manifest with [path] instead
          const options = loaderUtils.getOptions(loaderContext);
          pathName = loaderUtils.interpolateName(
            loaderContext,
            '[path]',
            // Use any custom context we've been given, otherwise just use the root context
            { context: options && options.context ? options.context : rootContext }
          );
        }
        
        let originalName = path.join(pathName, path.basename(module.userRequest));
```

Reply:
It's been a while since I looked at this code so I'd need to re-familarise myself, but I'd imagine the intention of the if-statement was because of the comment: `// Don't prefix any keys that already start with [path]`. I think I was trying to limit the diff so wanted to keep the initial assignment to `originalName` as similar to master as possible.

Your code looks simpler and seems like it achieves the same purpose by splitting the variables, so it's probably equivalent (assuming tests pass).